src/panoptes/pocs/camera/camera.py

```diff
@@ -68,28 +68,40 @@ class AbstractCamera(PanBase, metaclass=ABCMeta):
-        focuser (`panoptes.pocs.focuser.AbstractFocuser`|None): Focuser for the camera,
-        default None.
+        filter_type (str): Type of filter attached to camera. If a filterwheel is present this
+            will return the filterwheel.current_filter property, otherwise it will return the
+            value of the filter_type keyword argument, or if that argument was not given it
+            it will query the camera driver, e.g. 'M' for unfiltered monochrome camera,
+            'RGGB' for Bayer matrix colour camera.
+        focuser (`panoptes.pocs.focuser.AbstractFocuser`|None): Focuser for the camera, default
+            None.
         filter_wheel (`panoptes.pocs.filterwheel.AbstractFilterWheel`|None): Filter wheel for the
-        camera,
-            default None.
+            camera, default None.
+        uid (str): Unique identifier of the camera.
         is_primary (bool): If this camera is the primary camera for the system, default False.
         model (str): The model of camera, such as 'gphoto2', 'sbig', etc. Default 'simulator'.
         name (str): Name of the camera, default 'Generic Camera'.
         port (str): The port the camera is connected to, typically a usb device, default None.
+        temperature (astropy.units.Quantity): Current temperature of the image sensor.
         target_temperature (astropy.units.Quantity): image sensor cooling target temperature.
         temperature_tolerance (astropy.units.Quantity): tolerance for image sensor temperature.
+        cooling_enabled (bool): True if image snsor cooling is active.
+        cooling_power (float): Current image sensor cooling power level.
+        egain (astropy.units.Quantity): Image sensor gain in e-/ADU as reported by the camera.
         gain (int): The gain setting of the camera (ZWO cameras only).
+        bitdepth (astropy.units.Quantity): ADC bit depth in bits.
         image_type (str): Image format of the camera, e.g. 'RAW16', 'RGB24' (ZWO cameras only).
         timeout (astropy.units.Quantity): max time to wait after exposure before TimeoutError.
         readout_time (float): approximate time to readout the camera after an exposure.
         file_extension (str): file extension used by the camera's image data, e.g. 'fits'
         library_path (str): path to camera library, e.g. '/usr/local/lib/libfli.so' (SBIG, FLI, ZWO)
         properties (dict): A collection of camera properties as read from the camera.
+        is_connected (bool): True if camera is connected.
         is_cooled_camera (bool): True if camera has image sensor cooling capability.
         is_temperature_stable (bool): True if image sensor temperature is stable.
         is_exposing (bool): True if an exposure is currently under way, otherwise False.
+        is_ready (bool): True if the camera is ready to take an exposure.
+        can_take_internal_darks (bool): True if the camera can take internal dark exposures.
```

Comment:
Can we do `darks_enabled` to keep with the `cooling_enabled`?

My first thought was that wouldn't work because what we are sort of saying is we want every camera within POCS to be dark-capable by providing some mechanism (lens cap, slew to home, opaque filter) that allows the to take dark shots.  

However, I still think a `ZWO` camera class would have a `dark_enabled=False` by default, and then when you create particular instances of `ZWO` with an opaque filter, you then can set `dark_enabled=True` (could be set automatically on `init` by testing for filterwheel, etc).

This would change some of the structure below, but I think would be more consistent with the cooling feature of a camera (and again points to that plugin/decorator mechanism).

Reply:
I like the principle, but not entirely happy with the nomenclature.

From my point of view all cameras can take dark exposures, it's just that some need more help than others... It's (almost) always possible to have a human manually put on a lens cap, or equivalent. That's why I did this so that it's still possible to take an exposure with `dark=True` even when there is no supported mechanism for automatically blocking the light. In that case all that `dark=True` does is set the `IMAGETYP` FITS header accordingly, under the assumption that something external is taking care of keeping light out.

Reply:
Right, it's mostly the nomenclature that is a bit odd to me. I definitely like the idea the all cameras could somehow provide darks. I almost suggested something like `has_shutter_control` but then you also have to build some things around it to make it work like you want.

Let's not worry about it too much right now and maybe with some usage and implementation it will evolve.